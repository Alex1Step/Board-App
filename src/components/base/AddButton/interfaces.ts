export interface IaddButtonProps {
    className: string;
    text: string;
    type: string;
    onClick: () => void;
}