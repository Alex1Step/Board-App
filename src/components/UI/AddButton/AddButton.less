.Task {
  background: greenyellow;
  cursor: pointer;
}

.Board {
  background: fuchsia;
  width: 100px;
  height: 100px;
  cursor: pointer;
}
