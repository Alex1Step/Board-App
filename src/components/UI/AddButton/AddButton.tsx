import React from 'react';
import { Button } from 'antd';
import styles from './AddButton.less';

interface AddButtonProps {
    text: string;
    type: string;
    onClick: () => void;
}

const AddButton = (props: AddButtonProps) => {
    const cls = [styles[props.type]];

    return (
        <Button className={cls.join(' ')} onClick={props.onClick} type="primary">
            {props.text}
        </Button>
    );
};

export default AddButton;
