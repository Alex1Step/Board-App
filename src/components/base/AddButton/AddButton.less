.addButton {
  cursor: pointer;
}

.addBoard {
  width: 180px;
  margin-bottom: 15px;
}

.addCard {
  width: 130px;
}
