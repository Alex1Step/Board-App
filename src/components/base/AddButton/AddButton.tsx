import React from 'react';

import { Button } from 'antd';

import { IaddButtonProps } from './interfaces';

import cn from 'classnames';

import styles from './AddButton.less';

const AddButton = (props: IaddButtonProps): React.ReactElement => {
    return (
        <Button
            className={cn({
                [styles.addButton]: true,
                [styles.addBoard]: props.className === 'addBoard',
                [styles.addCard]: props.className === 'addCard',
            })}
            onClick={props.onClick}
            type="primary"
        >
            {props.text}
        </Button>
    );
};

export default AddButton;
