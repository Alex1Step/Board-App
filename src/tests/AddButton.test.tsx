Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
    })),
});

//with ENZYME

import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import AddButton from '../components/base/AddButton/AddButton';
import { Button } from 'antd';

describe('Button for adding components', () => {
    it('Rendering correctly', () => {
        const spy = jest.fn();
        const rend = renderer
            .create(<AddButton className="addBoard" text="text" type="primary" onClick={spy} />)
            .toJSON();
        expect(rend).toMatchSnapshot();
    });

    it('On click function works', () => {
        const spy = jest.fn();
        const btnElement = shallow(<AddButton className="addBoard" text="text" type="primary" onClick={spy} />);
        btnElement.find(Button).simulate('click');
        btnElement.find(Button).simulate('click');
        btnElement.find(Button).simulate('click');
        expect(spy).toHaveBeenCalled();
        expect(spy.mock.calls.length).toEqual(3);
    });
    //CLASSES
    it('Has classes', () => {
        const spy = jest.fn();
        const btnElement = shallow(<AddButton className="addBoard" text="text" type="primary" onClick={spy} />);
        // console.log(btnElement.find(Button).debug());
        // expect(btnElement.find(Button).hasClass('addButton')).toBeTruthy();
    });

    it('Has label text', () => {
        const spy = jest.fn();
        const btnElement = mount(<AddButton className="addBoard" text="text" type="primary" onClick={spy} />);
        expect(btnElement.find(Button).text()).toBe('text');
    });
});
