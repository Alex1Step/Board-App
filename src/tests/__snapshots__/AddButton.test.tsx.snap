// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button for adding components Rendering correctly 1`] = `
<button
  className="ant-btn ant-btn-primary addButton addBoard"
  onClick={[Function]}
  type="button"
>
  <span>
    text
  </span>
</button>
`;
